src/legacy/core_plugins/interpreter/public/functions/esaggs.ts

```diff
@@ -112,13 +112,18 @@ export const esaggs = (): ExpressionFunction<typeof name, Context, Arguments, Re
       queryFilter,
     });
 
-    return {
+    const table: KibanaDatatable = {
       type: 'kibana_datatable',
       rows: response.rows,
-      columns: response.columns.map(column => ({
+      columns: response.columns.map((column: any) => ({
         id: column.id,
         name: column.name,
+        formatterMapping: column.aggConfig.params.field
```

Comment:
This is not complete yet (there might be cases where this way of retrieving the formatters fails), but it works as a proof of concept for simple visualizations in Lens. As soon as we decide to follow this approach, I will clean up here